### DISCUSSION TITLE
Query editor: should we create a mechanism for specifying the default query for a datasource

### DISCUSSION THREAD

**Discussion by toddtreece** [grafana/grafana#43208] [Data Sources]
**What happened**:

It seems like properties from the previous data source query are being passed to the next datasource's query editor. For example, if I switch from the `-- Grafana --` data source to another data source, I get {refId: "A", queryType: "randomWalk"} in the `query` prop of the next data source. It seems like it [should be reset here][reset], but something else is modifying it.

When debugging, I noticed that it was [reset when it reached here][fine], but when stepping i'm seeing that after [this publish fires][publish], `this.targets` is modified with the properties from the previous data source's last query, but that is as far as I could track it.

**What you expected to happen**:

The new data source should not receive the query from the previous data source when switching data sources.

**How to reproduce it (as minimally and precisely as possible)**:

I'm using the `-- Grafana --` data source as an example, but this seems to happen when switching between any data sources.

1. Add a breakpoint or `console.log` to monitor the `query` prop in a QueryEditor component of a test data source.
2. Create a new panel, and select the `-- Grafana --` data source.
3. Select your test data source from the dropdown. You should see that `{refId: "A", queryType: "randomWalk"}` is passed to your data source's `query` property.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.1.0
- Data source type & version: Multiple data sources
- OS Grafana is installed on: Ubuntu
- User OS & Browser: macOS safari, chrome, & firefox
- Grafana plugins: 
- Others:

[reset]: https://github.com/grafana/grafana/blob/5986d99f51bb4b6fed381e18a706e6d6eb5dfad6/public/app/features/query/components/QueryGroup.tsx#L135-L136
[fine]: https://github.com/grafana/grafana/blob/5986d99f51bb4b6fed381e18a706e6d6eb5dfad6/public/app/features/dashboard/state/PanelModel.ts#L443
[publish]: https://github.com/grafana/grafana/blob/5986d99f51bb4b6fed381e18a706e6d6eb5dfad6/public/app/features/dashboard/state/PanelModel.ts#L446

**Comment by ashharrison90** [comment]
i've put a band-aid fix for the --Grafana-- datasource in the PR above ☝️ 
@toddtreece mentioned this was affecting other datasources. i suspect the underlying cause is similar.

what we really need is a mechanism for providing a default query for each datasource.
that way when changing datasource, the code [here](https://github.com/grafana/grafana/blob/5986d99f51bb4b6fed381e18a706e6d6eb5dfad6/public/app/features/query/components/QueryGroup.tsx#L135-L136) in `QueryGroup` can set the query to be the next datasource's default query. that will require a bit of architectural thinking.

i'm going to rename the issue and use this to track implementing that.

**Comment by pkolyvas** [comment]
As noted above @ashharrison90 fixed the part of this issue that was specifically a bug. We're going to bump this out to discussions to cover the larger question about default queries as an enhancement request.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
